apis/notification/v1alpha1/snssubscription_types.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SNSSubscriptionParameters define the desired state of a AWS SNS Topic
+type SNSSubscriptionParameters struct {
+
+	// TopicArn is the Arn of the SNS Topic
+	// +optional
+	TopicArn string `json:"topicArn,omitempty"`
+
+	// TopicArnRef references a SNS Topic and retrieves its TopicArn
+	// +optional
+	TopicArnRef *runtimev1alpha1.Reference `json:"topicArnRef,omitempty"`
+
+	// TopicArnSelector selects a reference to a SNS Topic and retrieves
+	// its TopicArn
+	// +optional
+	TopicArnSelector *runtimev1alpha1.Selector `json:"topicArnSelector,omitempty"`
+
+	// The subscription's protocol.
+	// +immutable
+	Protocol *string `json:"protocol"`
+
+	// The subscription's endpoint
+	// +immutable
+	Endpoint *string `json:"endpoint"`
```

Comment:
It appears that Endpoint and Protocol are not optional. If so, they should be strings and not pointers to string.